be/src/storage/index/vector/tenann/tenann_index_builder.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef WITH_TENANN
+
+#include "storage/index/vector/tenann/tenann_index_builder.h"
+
+#include <tenann/util/threads.h>
+
+#include "column/array_column.h"
+#include "common/config.h"
+#include "storage/index/vector/tenann/tenann_index_utils.h"
+#include "tenann/factory/index_factory.h"
+
+namespace starrocks {
+// =============== TenAnnIndexBuilderProxy =============
+
+Status TenAnnIndexBuilderProxy::init() {
+    ASSIGN_OR_RETURN(auto meta, get_vector_meta(_tablet_index, std::map<std::string, std::string>{}))
+
+    RETURN_IF_ERROR(success_once(_init_once, []() {
+                        tenann::OmpSetNumThreads(config::config_tenann_build_concurrency);
+                        return Status::OK();
+                    }).status());
+
+    CHECK(meta.common_params().contains("dim")) << "Dim is a critical common param";
```

Comment:
Consider do the error handling and return error status when the condition doesn't meet. This `CHECK` will crash the process.